src/position.rs

```diff
@@ -346,17 +348,24 @@ impl From<shogi_core::PartialPosition> for PartialPosition {
 
 #[derive(Debug, Clone)]
 struct State {
+    /// Zobrist hashes for (board ^ side, hand)
+    /// TODO: It seems it's always used by XOR-ing two keys. Can we only store a single XOR-ed key?
```

Comment:
Although I'm not using it specifically, I think that it would be easier to calculate the judgment when searching for positions with a "superiority relationship" if it is possible to independently obtain the Hash of the board only, not including the hand pieces, for example, when searching for Tsume-Shogi.

[詰将棋を解くアルゴリズムにおける優越関係の効率的な利用について](https://ipsj.ixsq.nii.ac.jp/ej/index.php?active_action=repository_view_main_item_detail&page_id=13&block_id=8&item_id=97465&item_no=1)

Reply:
nit: grammar
```suggestion
    /// TODO: It seems whenever we use this field its components (two keys) are always XOR-ed. Can we only store a single XOR-ed key?
```